Attacker detection, log cleaning and various fixes

mzap opened the pull request.

Commit:
feat(core): add attacker detection

with an env var created by the shell we spawn

Commit:
feat(core): add log cleaner/cleanup

ghost commented:
Added the cleanup functionnality, feel free to check it (my define are definitely wrong, they are already xor'd, we need to xor them via our conf file, in hidden literals)

Commit:
fix(core): add files in builder

streamline process, still a compile error to deal with
(I need unistd, that defines open, which is a defined hook)

Commit:
fix(core): fix open hook and compile errors

clean up of headers and files too, with a few comments along the way

Commit:
feat(core): add stealth mode to open

and hiding non-hooked functions

Commit:
fix(core): broken stealth on certain cases

Commit:
fix(core): UWTMP_FILE used

not really sure it's stealthy enough that way, we'll see

ghost commented:
It should be merge-ready now.
